spec/presenters/submission_presenter_spec.rb

```diff
@@ -1,68 +1,98 @@
 require 'rails_helper'
 
 RSpec.describe SubmissionPresenter do
-  subject { described_class.new(submission, rates, submissions_repository) }
-  let!(:rates) { submission.rates }
-  let!(:submissions_repository) { double }
+  let(:submission) { instance_double(Submission) }
+  let(:rates) { double }
+  let(:submission_repository) { double }
+  let(:submission_presenter) { described_class.new(submission, rates, submission_repository) }
 
-  describe do
-    let!(:submission) { FactoryGirl.build(:submission) }
+  describe "is a delegator" do
+    subject { submission_presenter }
 
     it "acts like a Submission" do
-      expect(subject).to eq(submission)
-      expect(subject).not_to be_a(Submission)
+      is_expected.to eq(submission)
+      is_expected.not_to be_a(Submission)
     end
   end
 
   describe "#average_rate" do
     describe "when submission is unrated" do
-      let!(:submission) { FactoryGirl.build(:submission) }
+      subject { submission_presenter.average_rate }
 
-      it { expect(subject.average_rate).to eq(nil) }
+      before { allow(submission).to receive(:rated?).and_return(false) }
+
+      it { is_expected.to eq(nil) }
     end
 
     describe "when submission is rated and has no rates" do
-      before do
-        allow(Setting).to receive(:get).and_return(
-          FactoryGirl.create(:setting, required_rates_num: 0))
-      end
+      subject { submission_presenter.average_rate }
 
-      let!(:submission) do
-        FactoryGirl.build(:submission, :rated, required_rates_num: 0)
+      before do
+        allow(submission).to receive(:rated?).and_return(true)
+        allow(rates).to receive(:count).and_return(0)
```

Comment:
I think you get a little bit too far with this mocking here. I mean, it's good to mock, but we need to be careful.
Here e.g. you are testing implementation of the private method (this `allow(rates).to receive(:count).and_return(0)`.
It would be better to write this test in a way that does not rely on implementation.
We know that we need to sum values from rates, so I would create 'stupid' rate objects, that have all they need to work with that method (so I guess only `value`?).
If it's possible to create mock object it's better to do this than have spaghetti of `allow`/`expect`s, cause it makes tests really fragile (to changes) and not so reliable (sometimes tests are passing but something is not working ; ]).